sdk/examples/xo_python/sawtooth_xo/processor/xo_state.py

```diff
@@ -0,0 +1,203 @@
+
+            except ValueError:
+                raise InternalError("Failed to deserialize game data.")
+
+        else:
+            game_list = {}
+            board = state = player1 = player2 = None
+
+        game = Game(board=board,
+                    state=state,
+                    player1=player1,
+                    player2=player2)
+
+        self._game = game_list
+        self._game[self._game_name] = game
+
+    def delete_game(self):
+        del self._game[self._game_name]
+        if self._game:
+            self._store_game()
+        else:
+            self._delete_game()
+
+    def _store_game(self):
+        state_data = '|'.join(sorted([
+            ','.join([name, g.board, g.state, g.player1, g.player2])
+            for name, g in self._game.items()
+        ])).encode()
+
+        addresses = self._context.set_state(
+            {self._game_address: state_data},
+            timeout=self.TIMEOUT)
+
+        if len(addresses) < 1:
+            raise InternalError("State set error: {}".format(
+                self._game_address))
+
+    def _delete_game(self):
+        addresses = self._context.delete_state(
+            [self._game_address],
+            timeout=self.TIMEOUT)
+
+        if len(addresses) < 1:
+            raise InternalError("State delete error: {}".format(
+                self._game_address))
+
+    def store_game(self):
+        """Store the game state in the validator state.
+        """
+
+        self._store_game()
+
+    def create_game(self):
+        """Create an empty board at the beginning of play.
+        """
+
+        self._game[self._game_name].board = "-" * 9
+        self._game[self._game_name].state = "P1-NEXT"
+        self._game[self._game_name].player1 = ""
+        self._game[self._game_name].player2 = ""
+
+    def update_board(self, board):
+        """Update the game state with new board state.
+
+        Args:
+            board (str): 9 characters representing the xo board.
+        """
+
+        self._game[self._game_name].board = board
+
+    def update_state(self, state):
+        """Update the variable tracking which player's turn it is.
+
+        Args:
+            state (str): The state of the game. e.g. P1-NEXT, P2-WIN
+        """
+
+        self._game[self._game_name].state = state
+
+    def create_player1(self, player1):
+
+        self._game[self._game_name].player1 = player1
+
+    def create_player2(self, player2):
+
+        self._game[self._game_name].player2 = player2
+
+    @property
+    def state(self):
+        """A string signifying the state the game is in before the current
+        transaction.
+
+        Notes:
+            - P1-WIN
+            - P2-WIN
+            - P1-NEXT
+            - P2-NEXT
+            - TIE
+
+        Returns (str): The state that the game is in.
+        """
+
+        return self._game[self._game_name].state
+
+    @property
+    def board(self):
+        """The game board before the current transaction.
+
+
```

Comment:
Missing `Returns` comment

Reply:
Thanks. I will update this after I get more reviews.